dali/python/nvidia/dali/types.py

```diff
@@ -157,74 +165,209 @@ def __init__(self, value, dtype=None):
+
+def _is_numpy_array(value):
+    if 'numpy.ndarray' in str(type(value)):
+        return True
+
+def ConstantNode(device, value, dtype, shape, layout):
+    data = value
+    if _is_numpy_array(value):
+        import numpy as np
+
+        # 64-bit types are not supported - downgrade the input
+        if value.dtype == np.float64:
+            value = value.astype(np.float32)
+        if value.dtype == np.int64:
+            value = value.astype(np.int32)
+        if value.dtype == np.uint64:
+            value = value.astype(np.uint32)
+
+        def _numpy_to_dali_type(t):
+            if t is None:
+                return None
+            if t == np.bool:
+                return DALIDataType.BOOL
+
+            if t == np.float16:
+                return DALIDataType.FLOAT16
+            if t == np.float32:
+                return DALIDataType.FLOAT
+            if t == np.float64:
+                return DALIDataType.FLOAT64
+
+            if t == np.uint8:
+                return DALIDataType.UINT8
+            if t == np.int8:
+                return DALIDataType.INT8
+            if t == np.uint16:
+                return DALIDataType.UINT16
+            if t == np.int16:
+                return DALIDataType.INT16
+            if t == np.uint32:
+                return DALIDataType.UINT32
+            if t == np.int32:
+                return DALIDataType.INT32
+            if t == np.uint64:
+                return DALIDataType.UINT64
+            if t == np.int64:
+                return DALIDataType.INT64
+            raise TypeError("Unsupported type: " + str(t))
+
+        actual_type = _numpy_to_dali_type(value.dtype)
+        if dtype is None:
+            dtype = actual_type
+        if shape is not None:
+            value = np.copy(value.reshape(shape), order='C')
+        else:
+            value = np.copy(value, order='C')
+            shape = value.shape
+        data = value.flatten().tolist()
+    else:
+        def _type_from_value_or_list(v):
+            if isinstance(v, (list, tuple)):
+                for x in v:
+                    if isinstance(x, float):
+                        return DALIDataType.FLOAT
+                return DALIDataType.INT32
+
+            if isinstance(v, float):
+                return DALIDataType.FLOAT
+            return DALIDataType.INT32
+
+        actual_type = _type_from_value_or_list(value)
+
+    import nvidia.dali.ops as ops
+
+    def _convert(x, type):
+        if isinstance(x, (list, tuple)):
+            return [type(y) for y in x]
+        return type(x)
+
+    isint = actual_type in _int_like_types
+    idata = _convert(data, int) if isint else None
+    fdata = None if isint else data
+    if device is None:
+        device = "cpu"
+
+    op = ops.Constant(device = device, fdata = fdata, idata = idata,
+                      shape = shape, dtype = dtype, layout = layout)
+    return op()
+
+def Constant(value, dtype = None, shape = None, layout = None, device = None):
```

Comment:
Maybe pass additional kwargs to the ops.Constant?
I'm thinking about naming the operator for example.

Reply:
Done.